ci-operator/step-registry/azure/provision/service-principal/minimal-permission/azure-provision-service-principal-minimal-permission-ref.yaml

```diff
@@ -19,18 +19,12 @@ ref:
     documentation: |-
       The provision way to set up cluster to test installation with Azure crendentials with minimal permissions.
       Valid value: IPI UPI
-  - name: ENABLE_MIN_PERMISSION_FOR_MARKETPLACE
-    default: "false"
+  - name: AZURE_MINIMAL_PERMISSIONS_USAGE
+    default: "cluster"
     documentation: |-
-      Assign addtional permissions for SP to install cluster with marketplace image
-  - name: ENABLE_MIN_PERMISSION_FOR_DES
-    default: "false"
-    documentation: |-
-      Assign additional permissions for SP to install cluster configured disk encryption set
-  - name: ENABLE_MIN_PERMISSION_FOR_STS
```

Comment:
Maybe we can keep this ENV, or else, we need to update many places beside job config yaml, also `ipi-conf-azure-oidc-creds-deprovision` and `ipi-conf-azure-oidc-creds-provision`. This also help keep consistent with gcp. AWS is using `AWS_CCOCTL_USE_MINIMAL_PERMISSIONS` ENV var to control. Or else, we would have 3 different ENV var to control the behavior.

Reply:
If keep env `ENABLE_MIN_PERMISSION_FOR_STS`, maybe need to introduce new env to control cluster sp creation, any thoughts?